modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsUpgradeProcess.java

```diff
@@ -0,0 +1,1074 @@
+			preparedStatement.setString(13, String.valueOf(increment()));
+			preparedStatement.setString(
+				14, DDMStructureConstants.VERSION_DEFAULT);
+			preparedStatement.setString(15, name);
+			preparedStatement.setString(16, description);
+			preparedStatement.setString(17, definition);
+			preparedStatement.setString(18, StorageType.DEFAULT.toString());
+			preparedStatement.setInt(19, DDMStructureConstants.TYPE_AUTO);
+			preparedStatement.setTimestamp(20, lastPublishDate);
+
+			preparedStatement.executeUpdate();
+
+			return structureId;
+		}
+	}
+
+	private void _addDDMStructureLayout(
+			long groupId, long companyId, long userId, String userName,
+			Timestamp createDate, Timestamp modifiedDate,
+			long structureVersionId, String name, String description,
+			String definition)
+		throws Exception {
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"insert into DDMStructureLayout (mvccVersion, ",
+					"ctCollectionId, uuid_, structureLayoutId, groupId, ",
+					"companyId, userId, userName, createDate, modifiedDate, ",
+					"classNameId, structureLayoutKey, structureVersionId, ",
+					"name, description, definition) values (0, 0, ?, ?, ?, ?, ",
+					"?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"))) {
+
+			preparedStatement.setString(1, PortalUUIDUtil.generate());
+			preparedStatement.setLong(2, increment());
+			preparedStatement.setLong(3, groupId);
+			preparedStatement.setLong(4, companyId);
+			preparedStatement.setLong(5, userId);
+			preparedStatement.setString(6, userName);
+			preparedStatement.setTimestamp(7, createDate);
+			preparedStatement.setTimestamp(8, modifiedDate);
+			preparedStatement.setLong(
+				9, PortalUtil.getClassNameId(DDMFormInstance.class));
+			preparedStatement.setString(10, String.valueOf(increment()));
+			preparedStatement.setLong(11, structureVersionId);
+			preparedStatement.setString(12, name);
+			preparedStatement.setString(13, description);
+			preparedStatement.setString(14, definition);
+
+			preparedStatement.executeUpdate();
+		}
+	}
+
+	private long _addDDMStructureVersion(
+			long groupId, long companyId, long userId, String userName,
+			Timestamp createDate, long structureId, String name,
+			String description, String definition, Timestamp statusDate)
+		throws Exception {
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"insert into DDMStructureVersion (mvccVersion, ",
+					"ctCollectionId, structureVersionId, groupId, companyId, ",
+					"userId, userName, createDate, structureId, version, ",
+					"parentStructureId, name, description, definition, ",
+					"storageType, type_, status, statusByUserId, ",
+					"statusByUserName, statusDate) values (0, 0, ?, ?, ?, ?, ",
+					"?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"))) {
+
+			long structureVersionId = increment();
+
+			preparedStatement.setLong(1, structureVersionId);
+
```

Comment:
@natocesarrego I think it's necessary because I'm initializing the variable `long structureVersionId = increment();` above and the`preparedStatement.setLong(1, structureVersionId);` depends on this variable.

I was also following the example from [DLFileEntryTypeDataDefinitionIdUpgradeProcess.java#L223-L235](https://github.com/liferay/liferay-portal/blob/08c6b85908e5745c20b53fd5c8e3032bbf15e165/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_0/DLFileEntryTypeDataDefinitionIdUpgradeProcess.java#L223-L235).